### PR TITLE
Fix docs: in labelname, convert to underscore

### PR DESCRIPTION
**Pull request by dapeleg-dn**
Fixes https://github.com/prometheus/docs/issues/735

### CODE REVIEW

File: docs/configuration/configuration.md [hunk 1]
```diff
@@ -37,7 +37,7 @@ Generic placeholders are defined as follows:
 * `<float>`: a floating-point number
 * `<host>`: a valid string consisting of a hostname or IP followed by an optional port number
 * `<int>`: an integer value
-* `<labelname>`: a string matching the regular expression `[a-zA-Z_][a-zA-Z0-9_]*`
+* `<labelname>`: a string matching the regular expression `[a-zA-Z_][a-zA-Z0-9_]*`. Any other special character in the source label is converted to an underscore. For example, the label `app.kubernetes.io/name` should be written as `app_kubernetes_io_name`.
```
Comment (roidelapluie) [inline docs/configuration/configuration.md:40]:
```suggestion
* `<labelname>`: a string matching the regular expression `[a-zA-Z_][a-zA-Z0-9_]*`. Any other special character in the source label should be converted to an underscore. For example, the label `app.kubernetes.io/name` should be written as `app_kubernetes_io_name`.
```

File: docs/configuration/configuration.md [hunk 2]
```diff
@@ -816,12 +816,12 @@ Available meta labels:
 * `__meta_docker_container_id`: the id of the container
 * `__meta_docker_container_name`: the name of the container
 * `__meta_docker_container_network_mode`: the network mode of the container
-* `__meta_docker_container_label_<labelname>`: each label of the container
+* `__meta_docker_container_label_<labelname>`: each label of the container, but you need to convert any dot, dash, and slash to an underscore
```
Comment (roidelapluie) [inline docs/configuration/configuration.md:819]:
```suggestion
* `__meta_docker_container_label_<labelname>`: each label of the container, with any unsupported characters converted to an underscore
```

**Review by roidelapluie — COMMENTED** [review]
We have reviewed this during our bug scrub and left some suggestions.

The note about special characters can be applied to the whole file.

Thank you for your contribution!

**Comment by dapeleg-dn** [comment]
Changed the text (almost) as suggested.

**Comment by roidelapluie** [comment]
Can we keep on with "unsupported" ? non alpha-numeric does not quite fit, as e.g. labels can not start with a numeric so that would be unsupported, and _ is not alpha numeric either

**Comment by dapeleg-dn** [comment]
> Can we keep on with "unsupported" ? non alpha-numeric does not quite fit, as e.g. labels can not start with a numeric so that would be unsupported, and _ is not alpha numeric either

Done.

**Review by jesusvazquez — APPROVED** [review]
Hello, this PR has been reviewed during the bug scrub.

Thank you for addressing the requested changes and improving Prometheus's docs 🙏 

Sorry for the big delay.

**Pull request merged**